File: Assets/Scripts/Weapons/AnimationSpeedController.js
var AnimationSpeeds : float[];

function Start ()
{
    var iteration = 0;
    for (var state : AnimationState in GetComponent.<Animation>())
    {
        
        transform.GetComponent.<Animation>()[state.name].speed = AnimationSpeeds[iteration];
        iteration += 1;
    }
    iteration = 0;
}

File: Assets/Scripts/Weapons/AnimationSpeedController.cs
using UnityEngine;
using System.Collections;

public class AnimationSpeedController : MonoBehaviour
{
    public float[] AnimationSpeeds;

    void Start (){
        int iteration = 0;
        foreach (AnimationState state in GetComponent<Animation>())
        {
            transform.GetComponent<Animation>()[state.name].speed = AnimationSpeeds[iteration];
            iteration += 1;
        }
        
    }
}